src/oge/column_checks.py

```diff
@@ -530,25 +778,50 @@ def apply_dtypes(df: pd.DataFrame) -> pd.DataFrame:
         pd.DataFrame: original data frame with type converted columns.
     """
     dtypes = get_dtypes()
-    datetime_columns = [
+    datetime_cols = [
+        "boiler_operating_date",
+        "boiler_retirement_date",
+        "current_planned_generator_operating_date",
         "datetime_utc",
         "datetime_local",
-        "report_date",
+        "emission_control_operating_date",
+        "emission_control_retirement_date",
         "generator_operating_date",
         "generator_retirement_date",
-        "current_planned_generator_operating_date",
+        "new_source_review_date",
+        "operating_datetime_utc",
+        "original_planned_generator_operating_date",
+        "other_modifications_date",
+        "planned_derate_date",
+        "planned_generator_retirement_date",
+        "planned_repower_date",
+        "planned_uprate_date",
         "plant_operating_date",
         "plant_retirement_date",
+        "report_date",
+        "uprate_derate_completed_date",
     ]
+
     cols_missing_dtypes = [
         col
         for col in df.columns
-        if (col not in dtypes) and (col not in datetime_columns)
+        if ((col not in dtypes) and (col not in datetime_cols))
     ]
     if len(cols_missing_dtypes) > 0:
         logger.warning(
             "The following columns do not have dtypes assigned in "
             "`column_checks.get_dtypes()`"
         )
         logger.warning(cols_missing_dtypes)
-    return df.astype({col: dtypes[col] for col in df.columns if col in dtypes})
+
+    # apply the dtype if the column is in the dtype dict
+    df = df.astype({col: dtypes[col] for col in df.columns if (col in dtypes)})
+    # apply datetime columns
+    for col in datetime_cols:
+        if col in df.columns:
+            try:
+                df[col] = df[col].astype("datetime64[s]")
+            except TypeError:
+                df[col] = df[col].dt.tz_localize(None).astype("datetime64[s]")
```

Comment:
When does this happen, i.e., that a column in `datetime_cols` cannot be converted to `datetime64[s]`? Just being curious

Reply:
This was an issue when there was a dataframe with a datetime_utc column that had already been made tzaware (eg the dtype is `datetime64[s. UTC]`